File: frontend/src/app/models/Appointment.ts
export class Appointment {
    static title: string;
    static start: Date;
    static authorId: number;
    static doctorId: number;

    constructor(public title: string,
                public start: Date,
                public authorId: number,
                public doctorId: number) {
        this.title = title;
        this.start = start;
        this.authorId = authorId;
        this.doctorId = doctorId;
    }

    static trueCopy(appointment: Appointment): Appointment {
        return new Appointment(
            Appointment.title,
            Appointment.start,
            Appointment.authorId,
            Appointment.doctorId,
            );
    }
}

File: backend/src/main/java/nl/hva/infosupport/models/Appointment.java
package nl.hva.infosupport.models;

import javax.persistence.*;
import java.time.Instant;
import java.util.Date;

@Entity
public class Appointment {
    @Id
    @GeneratedValue
    public int id;
    @Column(name = "title")
    public String title;
    @Column(name = "date")
    public Date start;
    @Column(name = "authorId")
    public int authorId;
    @Column(name = "doctorId")
    public int doctorId;

    public Appointment() {
    }

    public Appointment(String title, Date start, int authorId, int doctorId) {
        this.title = title;
        this.start = start;
        this.authorId = authorId;
        this.doctorId = doctorId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Date getStart() {
        return start;
    }

    public void setStart(Date start) {
        this.start = start;
    }

    public int getAuthorId() {
        return authorId;
    }

    public void setAuthorId(int authorId) {
        this.authorId = authorId;
    }

    public int getDoctorId() {
        return doctorId;
    }

    public void setDoctorId(int doctorId) {
        this.doctorId = doctorId;
    }
}
